public/app/core/utils/explore.ts

```diff
@@ -157,49 +157,72 @@ export function buildQueryTransaction(
 
 export const clearQueryKeys: (query: DataQuery) => object = ({ key, refId, ...rest }) => rest;
 
-const isMetricSegment = (segment: { [key: string]: string }) => segment.hasOwnProperty('expr');
+const metricProperties = ['expr', 'target'];
```

Comment:
this seems very specific to Prometheus and Graphite, a query model can contain any nested json object.